ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/TransactionLogBloomCacher.java

```diff
@@ -134,15 +126,19 @@ private long fillCacheFile(
       cachingStatus.currentBlock = blockNum;
       blockNum++;
     }
-    return blockNum - startBlock;
   }
 
-  public void cacheLogsBloomForBlockHeader(final BlockHeader blockHeader) {
+  public void cacheLogsBloomForBlockHeader(
+      final BlockHeader blockHeader,
+      final Optional<File> reusedCacheFile,
+      final boolean ensureChecks) {
     try {
       final long blockNumber = blockHeader.getNumber();
       LOG.debug("Caching logs bloom for block {}.", "0x" + Long.toHexString(blockNumber));
-      ensurePreviousSegmentsArePresent(blockNumber);
-      final File cacheFile = calculateCacheFileName(blockNumber, cacheDir);
+      if (ensureChecks) {
```

Comment:
Why did `ensureChecks` get introduced? Performance reasons?

Reply:
Yeah. At some places we don't want to do always this check